src/UI/Button.elm

```diff
@@ -1,342 +1,427 @@
+
 
+buttonWidth : Button msg -> Attribute msg
+buttonWidth (Button _ { width }) =
+    if width == WidthFull then
+        Element.width Element.fill
 
-colorAttrs : Options msg -> List (Attr () msg)
-colorAttrs { colorScheme, buttonType, isDisabled } =
+    else
+        Element.width Element.shrink
+
+
+buttonPadding : Button msg -> Attribute msg
+buttonPadding (Button { body } _) =
+    case body of
+        BodyText _ ->
+            Element.paddingXY 32 12
+
+        BodyIcon _ ->
+            Element.paddingXY 10 12
+
+        BodyEl _ ->
+            Element.paddingXY 10 12
+
+
+type alias ColorIntermediary =
+    ( ( Color, Color ), Maybe ( Color, Color ) )
+
+
+styleAttrs : Button msg -> List (Attribute msg)
+styleAttrs ((Button { click } { style }) as btn) =
     let
-        attrs normal hover =
-            case buttonType of
-                Link _ ->
-                    [ Font.color hover.text
-                    , Font.underline
-                    , mouseOver
-                        [ Font.color Theme.black
-                        ]
-                    ]
-
-                _ ->
-                    [ Background.color normal.bg
-                    , Font.color normal.text
-                    , mouseOver
-                        [ Background.color hover.bg
-                        , Font.color hover.text
-                        , Border.color hover.border
-                        ]
-                    , Border.color normal.border
-                    , Attributes.transition 100 [ "color", "background-color" ]
-                    ]
-
-        disabledMode =
-            attrs disabledColorScheme.normal disabledColorScheme.hover
-
-        enabledMode =
-            case buttonType of
-                Normal _ ->
-                    attrs colorScheme.normal colorScheme.hover
-
-                Link _ ->
-                    attrs colorScheme.normal colorScheme.hover
-
-                Toggle _ isEnabled ->
-                    if isEnabled then
-                        attrs colorScheme.normal colorScheme.hover
-
-                    else
-                        attrs colorScheme.hover colorScheme.normal
+        ( ( active, passive ) as normal, maybeHover ) =
+            colorHelper btn
+
+        ( hoverActive, hoverPassive ) =
+            Maybe.withDefault normal maybeHover
     in
-    ifThenElse isDisabled disabledMode enabledMode
+    case ( click, style ) of
+        ( ClickHref _, _ ) ->
+            [ Font.underline
+            , Font.color passive
+            , Element.mouseOver
+                [ Font.color hoverPassive
+                ]
+            ]
+
+        ( _, StyleFilled ) ->
+            [ Background.color active
+            , Font.color passive
+            , Element.mouseOver
+                [ Background.color hoverActive
+                , Font.color hoverPassive
+                ]
+            ]
+                ++ Element.colorTransition 100
+
+        ( _, StyleOutlined ) ->
```

Comment:
I think there is a size difference between the outline and the normal button because of the border size. So maybe we should have a border in the normal button as well.